helix-term/src/commands/lsp.rs

```diff
@@ -1398,3 +1405,120 @@ fn compute_inlay_hints_for_view(
 
     Some(callback)
 }
+
+pub fn compute_color_swatches_for_all_views(editor: &mut Editor, jobs: &mut crate::job::Jobs) {
+    if !editor.config().lsp.display_color_swatches {
+        return;
+    }
+
+    for (view, _) in editor.tree.views() {
+        let doc = match editor.documents.get(&view.doc) {
+            Some(doc) => doc,
+            None => continue,
+        };
+        if let Some(callback) = compute_color_swatches_for_view(view, doc) {
+            jobs.callback(callback);
+        }
+    }
+}
+
+fn compute_color_swatches_for_view(
+    view: &View,
+    doc: &Document,
+) -> Option<std::pin::Pin<Box<impl Future<Output = Result<crate::job::Callback, anyhow::Error>>>>> {
+    let view_id = view.id;
+    let doc_id = view.doc;
+
+    let language_server = doc
+        .language_servers_with_feature(LanguageServerFeature::ColorProvider)
+        .next()?;
```

Comment:
I would gracefully handle the error, maybe logging (info level) that there is no LSP that supports colour provider for the current file type and just do nothing, instead of propagating error

Reply:
If we do that then it will *always* show an error e.g. when we go into a python or a rust file, since those LSPs never have color providers

Reply:
I am not sure I understand. I will need to look over when I get back to a computer, maybe I'm missing something.

Reply:
> I am not sure I understand. I will need to look over when I get back to a computer, maybe I'm missing something.

From what I can tell, Helix tries to run every LSP method when it launches a language server. So if you mostly work with Rust, it will log that info message every time. this will clutter up the log. Other methods don't log anything if the language server doesn't provide them, since there is nothing we can do